### PR TITLE
[now-next] 404 Known Static Files Before Dynamic Routes

### PR DESCRIPTION
**Pull request by dav-is**
If someone uses a wildcard in the root of pages, it could collide with internal `_next` files:
```
pages/
  [slug]/
    [slug]/
      [slug]/
        [slug].js
```

For example:
`/_next/static/runtime/webpack-hash.js` exists and has no problem routing
`/_next/static/runtime/webpack-old-hash.js` doesn't exist on the server any more but would route to `/[slug].js` with status 200

This could cause strange edge cases related to loading a new version of a Next.js site along with the possible cost of execution if it's a lambda plus the cost of however large the response is.

# Side Notes

Should this:
https://github.com/zeit/now/blob/92d9f2d809b7f9abc767572125fa6eecc6468c16/packages/now-next/src/index.ts#L839

Instead be this?
```
 src: path.join('/', entryDirectory, '_next/static/(?:[^/]+/pages|chunks|runtime|css|media)/.+'),
```

If this is the case, I need to adjust my PR to do the same

### CODE REVIEW

File: packages/now-next/src/index.ts [hunk 1]
```diff
@@ -845,6 +845,7 @@ export const build = async ({
       // Next.js page lambdas, `static/` folder, reserved assets, and `public/`
       // folder
       { handle: 'filesystem' },
+      { src: '/_next(?:/.*)?', status: 404, dest: '/_error' },
```
Comment (Timer) [inline packages/now-next/src/index.ts]: Shouldn't this be moved to before `{ handle: 'filesystem' }` and just be `{ src: '/_next(.*)' }`?

**Review by Timer — CHANGES_REQUESTED** [review]

File: packages/now-next/src/index.ts (hunk 1 above)
Comment (Timer) [reply packages/now-next/src/index.ts]: Actually, there's dynamic data routes which exist under the `/_next` path. We can't route this to 404. I'm not sure why tests are passing.

Comment (dav-is) [reply packages/now-next/src/index.ts]: No because if it 404s before `{ handle: 'filesystem' }` it will still continue to route

Comment (dav-is) [reply packages/now-next/src/index.ts]: > dynamic data routes

What's the pattern?

Comment (Timer) [reply packages/now-next/src/index.ts]: `dynamicDataRoutes`

Comment (dav-is) [reply packages/now-next/src/index.ts]: I'm asking what the path pattern is for those (e.g. `/_next/dynamic/whatever`)

Comment (Timer) [reply packages/now-next/src/index.ts]: Yeah, GitHub only showed your question after I already posted that.

It looks like this:
```
`^\\/_next\\/data\\/${escapedBuildId}\\/blog\\/([^\\/]+?)\\.json$`
```

**Review by Timer — CHANGES_REQUESTED** [review]
I'm still not following the case this is solving for. The provided example is invalid, as `[slug].js` only matches **one level**. Could you be a bit more specific?

**Comment by dav-is** [comment]
@Timer That's good to know, I didn't realize that. I guess this is more of an edge case than I thought.

Investigating closer, zeit.co uses `[teamSlug]/[project]/[id]/[suffix]`

**Review by Timer — CHANGES_REQUESTED** [review]

**Review by Timer — APPROVED** [review]
Seems legit. Thanks!

**Pull request merged**